src/scripts/windows/build.cmd

```diff
@@ -48,7 +50,9 @@ goto :init
     if /i "%~1"=="-b"               set "BUILD_TYPE=%~2"   & shift & shift & goto :parse
     if /i "%~1"=="--build-type"     set "BUILD_TYPE=%~2"   & shift & shift & goto :parse
 
-    if /i "%~1"=="--build-version"     set "BUILD_VERSION=%~2"   & shift & shift & goto :parse
+    if /i "%~1"=="--build-arch"     set "BUILD_ARCH=%~2"   & shift & shift & goto :parse
```

Comment:
Can you describe how this is intended to be used? This script is (was?) intnded to be used from a VsDevCmd command prompt. You specify the target and host architecture when you start the command prompt via the -arch and -host_arch command line parameters. How does --build-arch interact with the VsDevCmd settings

Reply:
This allows you to override the default behavior so automated builds can explicitly invoke x86 / x64 regardless of the dev command prompt flavor.

Reply:
as it happens, I ended up invoking both arches in the natvis cmakelists, so this change is largely superfluous

Reply:
So lets back this out

Reply:
If we ever wanted to build x64 from a build slave, how would we do that?

Reply:
Since it's purely an override, I'll go ahead and leave it in

Reply:
I don't think you can do that in cmake. For command line, you have to be in a VsDevCmd prompt so you have to choose your host and target archtiectures ahead of time. For VS, you have to select a specific configuration that includes target archtiecture. See docs on [CMakeSettings inheritEnvironments](https://docs.microsoft.com/en-us/cpp/ide/cmake-tools-for-visual-cpp?view=vs-2017#inherited-environments-visual-studio-2017-version-155)

Reply:
can you elaborate?  this change is simply treating the VSCMD_ARG_TGT_ARCH env var as a default

Reply:
My point is that you can't override the target archtiecture of the cmake build with this argument. It will still pick up the target arch from whatever settings the developer chose when they launched the VsDevCmd prompt. So if you're in an x86 arch prompt, this argument makes it look like you can build for x64, but you can't.

Also, it looks like XLANG_BUILD_ARCH is only used in two places
* src/package/natvis/CMakeLists.txt adds it to the cppwinrt_natvis_dir path. But doesn't that cmake file ALWAYS build x86 and x64? Why does cppwinrt_natvis_dir need to include the chosen XLANG_BUILD_ARCH in the path if it's going to build both anyway?
* src/package/vsix/CMakeLists.txt uses it for the MSBuild /p:Platform value, but that entire build script is predicated on XLANG_BUILD_ARCH being x86. I'm pretty sure you could use [CMAKE_LIBRARY_ARCHITECTURE](https://cmake.org/cmake/help/v3.9/variable/CMAKE_LIBRARY_ARCHITECTURE.html) instead.

Reply:
ok, I'll back this out

Reply:
👍